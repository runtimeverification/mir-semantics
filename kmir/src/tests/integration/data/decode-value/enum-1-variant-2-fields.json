{
  "bytes": [
    255,
    0,
    1,
    0
  ],
  "types": [
    [
      2,
      {
        "PrimitiveType": {
          "Int": "I16"
        }
      }
    ],
    [
      1,
      {
        "PrimitiveType": "Bool"
      }
    ]
  ],
  "typeInfo": {
    "EnumType": {
      "name": "Foo",
      "adt_def": 1,
      "discriminants": [
        0
      ],
      "fields": [
        [
          2,
          1
        ]
      ],
      "layout": {
        "fields": {
          "Arbitrary": {
            "offsets": [
              {
                "num_bits": 0
              },
              {
                "num_bits": 16
              }
            ]
          }
        },
        "variants": {
          "Single": {
            "index": 0
          }
        },
        "abi": {
          "ScalarPair": {}
        },
        "abi_align": 2,
        "size": {
          "num_bits": 32
        }
      }
    }
  }
}
