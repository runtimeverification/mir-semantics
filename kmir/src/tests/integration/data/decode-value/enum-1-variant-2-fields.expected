Aggregate ( variantIdx ( 0 ) , ListItem ( Integer ( 255 , 16 , true ) )
ListItem ( BoolVal ( true ) ) )
